### PR TITLE
Add Auto restart

### PR DESCRIPTION
**Pull request by 38elements**
Change:
* When the debug flag is enable and the source code is changed,  the server will automatically restart.

This is related #168.

### CODE REVIEW

File: sanic/__main__.py [hunk 1]
```diff
@@ -10,23 +10,70 @@
...
-        app = getattr(module, app_name, None)
-        if type(app) is not Sanic:
-            raise ValueError("Module is not a Sanic app, it is a {}.  "
-                             "Perhaps you meant {}.app?"
-                             .format(type(app).__name__, args.module))
-
-        app.run(host=args.host, port=args.port,
-                workers=args.workers, debug=args.debug)
+        if args.debug and not args.no_restart:
+            import os
```
Comment (seemethere) [inline sanic/__main__.py]: Can we break the imports out and put them at the top?

File: sanic/__main__.py [hunk 2]
```diff
@@ -10,23 +10,70 @@
...
+            import sys
+            import time
+            from subprocess import Popen
+            from signal import SIGTERM
+            from watchdog.observers import Observer
+            from watchdog.events import PatternMatchingEventHandler
+
+            class RestartHandler(PatternMatchingEventHandler):
```
Comment (seemethere) [inline sanic/__main__.py]: Can we also break this class declaration out of the conditional?

Comment (38elements) [inline sanic/__main__.py]: @seemethere 
Thank you for pointing it out.
Import and class declaration moved to the top.

File: sanic/__main__.py (hunk 1 above)
Comment (drinksober) [inline sanic/__main__.py]: in the class LiveLoadingHandler, can we add a method on_modified, on which we can listen the modified event and get the src_path, and the just reload the module of src_path

just like that 

    def on_modified(self, event):
        module = path and _find(event.src_path)
        if hasattr(module, '__spec__'):
            from importlib import reload
            app.router.reload()
            reload(module)
            reload(sanic)

def _find(path):
    try:
        return import_module(path)
    except (ImportError, TypeError):
        pass
    path = '.'.join(path.split('.')[:-1])
    if '.' in path:
        return _find(path)
    return path

**Comment by istommao** [comment]
Does this implementation (watchdog) merge into the master branch?

**Pull request closed without merging**